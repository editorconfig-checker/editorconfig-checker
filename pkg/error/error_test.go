package error

import (
	"errors"
	"testing"

	"github.com/editorconfig-checker/editorconfig-checker/v2/pkg/config"
)

func TestGetErrorCount(t *testing.T) {
	count := GetErrorCount([]ValidationErrors{})
	if count != 0 {
		t.Error("Expected empty slice to have no errors, got", count)
	}

	input := []ValidationErrors{
		{
			FilePath: "some/path",
			Errors: []ValidationError{
				{
					LineNumber: 1,
					Message:    errors.New("WRONG"),
				},
			},
		},
	}

	count = GetErrorCount(input)
	if count != 1 {
		t.Error("Expected one error slice to have exactly one erorr errors, got", count)
	}

	input = []ValidationErrors{
		{
			FilePath: "some/path",
			Errors: []ValidationError{
				{
					LineNumber: 1,
					Message:    errors.New("WRONG"),
				},
			},
		},
		{
			FilePath: "some/other/path",
			Errors: []ValidationError{
				{
					LineNumber: 1,
					Message:    errors.New("WRONG"),
				},
			},
		},
	}

	count = GetErrorCount(input)
	if count != 2 {
		t.Error("Expected two error slice to have exactly one erorr errors, got", count)
	}

	input = []ValidationErrors{
		{
			FilePath: "some/path",
			Errors: []ValidationError{
				{
					LineNumber: 1,
					Message:    errors.New("WRONG"),
				},
			},
		},
		{
			FilePath: "some/other/path",
			Errors: []ValidationError{
				{
					LineNumber: 1,
					Message:    errors.New("WRONG"),
				},
				{
					LineNumber: 2,
					Message:    errors.New("WRONG"),
				},
			},
		},
	}

	count = GetErrorCount(input)
	if count != 3 {
		t.Error("Expected three error slice to have exactly one erorr errors, got", count)
	}

	input = []ValidationErrors{
		{
			FilePath: "some/path",
			Errors:   []ValidationError{},
		},
		{
			FilePath: "some/other/path",
			Errors: []ValidationError{
				{
					LineNumber: 1,
					Message:    errors.New("WRONG"),
				},
				{
					LineNumber: 2,
					Message:    errors.New("WRONG"),
				},
			},
		},
	}

	count = GetErrorCount(input)
	if count != 2 {
		t.Error("Expected three error slice to have exactly one erorr errors, got", count)
	}
}

func TestPrintErrors(t *testing.T) {
	input := []ValidationErrors{
		{
			FilePath: "some/path",
			Errors:   []ValidationError{},
		},
		{
			FilePath: "/proc/cpuinfo",
			Errors: []ValidationError{
				{
					LineNumber: 1,
					Message:    errors.New("WRONG"),
				},
			},
		},
		{
			FilePath: "/proc/cpuinfoNOT",
			Errors: []ValidationError{
				{
					LineNumber: 1,
					Message:    errors.New("WRONG"),
				},
			},
		},
		{
			FilePath: "some/other/path",
			Errors: []ValidationError{
				{
					LineNumber: 1,
					Message:    errors.New("WRONG"),
				},
				{
					LineNumber: -1,
					Message:    errors.New("WRONG"),
				},
			},
		},
	}

	// wannabe test
	config1 := config.Config{}
	PrintErrors(input, config1)

	config2 := config.Config{Format: "gcc"}
	PrintErrors(input, config2)
}
