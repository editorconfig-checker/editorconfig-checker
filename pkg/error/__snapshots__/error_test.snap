
[TestFormatErrors - 1]
[]logger.LogMessage{
    {Level:"warning", Message:"proc/cpuinfo:"},
    {Level:"error", Message:"\t1: WRONG"},
    {Level:"warning", Message:"proc/cpuinfoNOT:"},
    {Level:"error", Message:"\t1: WRONG"},
    {Level:"warning", Message:"some/other/path:"},
    {Level:"error", Message:"\tWRONG"},
    {Level:"error", Message:"\t1: WRONG"},
    {Level:"warning", Message:"some/file/with/consecutive/errors:"},
    {Level:"error", Message:"\tfile-level error"},
    {Level:"error", Message:"\t1-2: message kind one"},
    {Level:"error", Message:"\t4: message kind one"},
    {Level:"error", Message:"\t5: message kind two"},
    {Level:"error", Message:"\t6: message kind one"},
}
---

[TestFormatErrors - 2]
[]logger.LogMessage{
    {Level:"error", Message:"proc/cpuinfo:1:0: error: WRONG"},
    {Level:"error", Message:"proc/cpuinfoNOT:1:0: error: WRONG"},
    {Level:"error", Message:"some/other/path:1:0: error: WRONG"},
    {Level:"error", Message:"some/other/path:0:0: error: WRONG"},
    {Level:"error", Message:"some/file/with/consecutive/errors:1:0: error: message kind one"},
    {Level:"error", Message:"some/file/with/consecutive/errors:2:0: error: message kind one"},
    {Level:"error", Message:"some/file/with/consecutive/errors:4:0: error: message kind one"},
    {Level:"error", Message:"some/file/with/consecutive/errors:5:0: error: message kind two"},
    {Level:"error", Message:"some/file/with/consecutive/errors:6:0: error: message kind one"},
    {Level:"error", Message:"some/file/with/consecutive/errors:0:0: error: file-level error"},
}
---

[TestFormatErrors - 3]
[]logger.LogMessage{
    {Level:"error", Message:"::error file=proc/cpuinfo,line=1::WRONG"},
    {Level:"error", Message:"::error file=proc/cpuinfoNOT,line=1::WRONG"},
    {Level:"error", Message:"::error file=some/other/path::WRONG"},
    {Level:"error", Message:"::error file=some/other/path,line=1::WRONG"},
    {Level:"error", Message:"::error file=some/file/with/consecutive/errors::file-level error"},
    {Level:"error", Message:"::error file=some/file/with/consecutive/errors,line=1,endLine=2::message kind one"},
    {Level:"error", Message:"::error file=some/file/with/consecutive/errors,line=4::message kind one"},
    {Level:"error", Message:"::error file=some/file/with/consecutive/errors,line=5::message kind two"},
    {Level:"error", Message:"::error file=some/file/with/consecutive/errors,line=6::message kind one"},
}
---
